### PR TITLE
Add Plugin ID , change from sqlite-analyzer to com.novoda.sqlite-analyzer

### PR DESCRIPTION
**Pull request by friedger**
This PR adds the publish-plugin from gradle.com (com.gradle.plugin-publish). This allows to publish the plugin at plugins.gradle.org. The plugin has been used for this plugin. In the future however, we will publish only via bintray (with the pull mechanism from the gradle portal). The publish-plugin provides the `pluginBundle` configuration.

The plugin is now available at https://plugins.gradle.org/plugin/com.novoda.sqlite-analyzer

This implies a change of the plugin id from sqlite-analyzer to com.novoda.sqlite-analyzer.
The old plugin id is deprecated.

### CODE REVIEW

File: analyzer/build.gradle [hunk 1]
```diff
@@ -38,3 +43,21 @@ publish {
...
+
+    plugins {
+        sqliteAnalyzerPlugin {
+            id = 'com.novoda.sqlite-analyzer'
+            displayName = 'SQLite Analyzer plugin'
+        }
+    }
+    
+    mavenCoordinates {
+      artifactId = "sqlite-analyzer"
```
Comment (stefanhoth) [inline analyzer/build.gradle]: is that enough or should the group be added here as well?

**Comment by stefanhoth** [comment]
Good effort. I wonder if the name change of the properties will have any side effect.

File: analyzer/build.gradle (hunk 1 above)
Comment (friedger) [inline analyzer/build.gradle]: This only overwrite the default values, here the artifaceId "analyzer"

Comment (stefanhoth) [inline analyzer/build.gradle]: Ok :)

**Comment by friedger** [comment]
Users of the library have to update their gradle files if they want to update to the next version. We can also continue to publish the old name on jcenter if we want that. Probably not worth the effort. In particular as it is nowadays standard to have namespaced plugin ids.

**Comment by blundell** [comment]
Why is it wanted that the plugin sits on `plugins.gradle.org` what is wrong with BinTray / JCenter?

**Comment by friedger** [comment]
This is not exclusive, however, the accessibility and visibility for gradle plugins is bigger at plugins.gradle.org than on bintray.

**Comment by devisnik** [comment]
Could we keep the old plugin properties as well, such that the old name would still resolve?

File: analyzer/build.gradle [hunk 2]
```diff
@@ -38,3 +43,21 @@ publish {
     description = 'Code generation for Java/Android database access.'
     website = 'https://github.com/novoda/sqlite-analyzer'
 }
+
+pluginBundle {
```
Comment (xrigau) [inline analyzer/build.gradle]: pluginBundle ??

Comment (friedger) [inline analyzer/build.gradle]: @xrigau this is the publish plugin from the gradle portal (com.gradle.plugin-publish). We have used it for this plugin but then decided to go the "old route" via bintray. I wanted to leave it as an example. I have update the description.

**Comment by stefanhoth** [comment]
@friedger Should this also be moved to plugindev as in novoda/gradle-android-command-plugin#77 ?

**Comment by friedger** [comment]
@stefanhoth I want to leave it like this for now until we exercised the plugindev plugin for one project.

**Comment by stefanhoth** [comment]
@friedger Leave like this or close the PR?

**Comment by friedger** [comment]
Please merge
Am 25.06.2015 11:15 schrieb "Stefan Hoth" <notifications@github.com>:

**Comment by stefanhoth** [comment]
@friedger There was another question by @devisnik regarding backwards-compatibility.

**Comment by friedger** [comment]
The old ID remains usable
Am 25.06.2015 14:17 schrieb "Stefan Hoth" <notifications@github.com>:

**Comment by stefanhoth** [comment]
Ok then. :ship:ing in 10mins.

**Comment by stefanhoth** [comment]
Or two hours...

**Pull request merged**